tests/integration/Core/Checkout/Document/Service/DocumentGeneratorTest.php

```diff
@@ -973,6 +973,54 @@ public static function readDocumentDataProvider(): \Generator
         ];
     }
 
+    public function testReadDocumentWithCapsMediaFileExtension(): void
+    {
+        $documentId = Uuid::randomHex();
+        $mediaId = Uuid::randomHex();
+
+        $fileSystem = static::getContainer()->get('shopware.filesystem.private');
+        $documentType = static::getContainer()
+            ->get('document_type.repository')
+            ->search(
+                (new Criteria())->addFilter(new EqualsFilter('technicalName', InvoiceRenderer::TYPE)),
+                $this->context
+            )
+            ->first();
+        static::assertInstanceOf(DocumentTypeEntity::class, $documentType);
+
+        $this->documentRepository->create([[
+            'id' => $documentId,
+            'documentTypeId' => $documentType->getId(),
+            'fileType' => PdfRenderer::FILE_EXTENSION,
+            'orderId' => $this->orderId,
+            'orderVersionId' => Defaults::LIVE_VERSION,
+            'config' => ['documentNumber' => '1001'],
+            'deepLinkCode' => 'dfr',
+            'static' => true,
+            'documentMediaFile' => [
+                'id' => $mediaId,
+                'mimeType' => 'application/pdf',
+                'fileExtension' => 'PDF',
+                'fileName' => 'textFileWithExtension',
+                'fileSize' => 1024,
+                'private' => true,
+                'mediaType' => new BinaryType(),
+                'uploadedAt' => new \DateTimeImmutable('2011-01-01T15:03:01.012345Z'),
+            ],
+        ]], $this->context);
+
+        $criteria = (new Criteria([$documentId]))->addAssociation('documentMediaFile');
+        /** @var DocumentEntity $document */
```

Comment:
please do not use var annotations. Especially in tests you can simply assert the type. also see https://github.com/shopware/shopware/blob/trunk/coding-guidelines/core/writing-code-for-static-analysis.md